jobsdb/jobsdb.go

```diff
@@ -3119,14 +3122,159 @@ func (jd *HandleT) getBackUpQuery(backupDSRange *dataSetRangeT, isJobStatusTable
-			where %[2]s.job_state in ('%[5]s', '%[6]s') order by  %[4]s.custom_val, %[2]s.job_id, %[2]s.exec_time asc limit %[7]d offset %[8]d) AS failed_jobs`, backupDSRange.ds.JobStatusTable, "job_status", backupDSRange.ds.JobTable, "job",
-			Failed.State, Aborted.State, backupRowsBatchSize, offset)
+		stmt = fmt.Sprintf(
+			`SELECT 
+				COALESCE(
+					json_agg(
+						json_build_object(
+							'job_id',
+							failed_jobs.job_id,
+							'workspace_id',
+							failed_jobs.workspace_id,
+							'uuid',
+							failed_jobs.uuid,
+							'user_id',
+							failed_jobs.user_id,
+							'parameters',
+							failed_jobs.parameters,
+							'custom_val',
+							failed_jobs.custom_val,
+							'event_payload',
+							failed_jobs.event_payload,
+							'event_count',
+							failed_jobs.event_count,
+							'created_at',
+							failed_jobs.created_at,
+							'expire_at',
+							failed_jobs.expire_at,
+							'id',
+							failed_jobs.id,
+							'job_id',
+							failed_jobs.status_job_id,
+							'job_state',
+							failed_jobs.job_state,
+							'attempt',
+							failed_jobs.attempt,
+							'exec_time',
+							failed_jobs.exec_time,
+							'retry_time',
+							failed_jobs.retry_time,
+							'error_code',
+							failed_jobs.error_code,
+							'error_response',
+							failed_jobs.error_response,
+							'parameters',
+							failed_jobs.status_parameters
+						)
+					), 
+					'[]' :: json
+				), 
+				count(*) 
+			FROM 
+				(
+				SELECT 
+					* 
+				FROM 
+					(
+					SELECT *, 
+					sum(
+					pg_column_size(jobs.event_payload)
+					) OVER (
+					ORDER BY 
+						jobs.custom_val, 
+						jobs.status_job_id, 
+						jobs.exec_time
+					) AS running_payload_size 
+					FROM
+						(
+						SELECT 
+							job.job_id, 
+							job.workspace_id, 
+							job.uuid, 
+							job.user_id, 
+							job.parameters, 
+							job.custom_val, 
+							job.event_payload, 
+							job.event_count, 
+							job.created_at, 
+							job.expire_at, 
+							job_status.id, 
+							job_status.job_id AS status_job_id, 
+							job_status.job_state, 
+							job_status.attempt, 
+							job_status.exec_time, 
+							job_status.retry_time, 
+							job_status.error_code, 
+							job_status.error_response, 
+							job_status.parameters AS status_parameters
+						FROM 
+							"%[1]s" % [2]s 
+							INNER JOIN "%[3]s" % [4]s ON % [2]s.job_id = % [4]s.job_id 
+						WHERE 
+							% [2]s.job_state IN ('%[5]s', '%[6]s') 
+						ORDER BY 
+							% [4]s.custom_val, 
+							% [2]s.job_id, 
+							% [2]s.exec_time ASC
+						LIMIT 
+							%[7]d
+						OFFSET
+							%[8]d
+						) jobs
+					) subquery 
+				WHERE 
+					subquery.running_payload_size <= %[9]d
+				) AS failed_jobs
+		  `, backupDSRange.ds.JobStatusTable, "job_status", backupDSRange.ds.JobTable, "job", Failed.State, Aborted.State, backupRowsBatchSize, offset, backupMaxTotalPayloadSize)
```

Comment:
since we are already using the `job` and `job_status` table aliases in the query, we shouldn't pass them as parameters